cranelift/isle/isle/src/printer.rs

```diff
@@ -17,234 +16,207 @@ where
+            Pattern::ConstPrim { val, .. } => format!("${}", &val.to_doc()),
+            Pattern::Wildcard { .. } => String::from("_"),
+            Pattern::Term { sym, args, .. } => {
+                // sexp(
+                //     Vec::from([sym.to_doc()])
+                //         .into_iter()
+                //         .chain(args.iter().map(|f| f.to_doc()))
+            }
+            Pattern::And { subpats, .. } => {
+                // sexp(
+                //     Vec::from([RcDoc::text("and")])
+                //         .into_iter()
+                //         .chain(subpats.iter().map(|p| p.to_doc()))
+            }
             Pattern::MacroArg { .. } => unimplemented!("macro arguments are for internal use only"),
         }
     }
 }
 
 impl Printable for IfLet {
-    fn to_doc(&self) -> RcDoc<()> {
-        // TODO(mbm): `if` shorthand when pattern is wildcard
-        sexp(vec![
-            RcDoc::text("if-let"),
-            self.pattern.to_doc(),
-            self.expr.to_doc(),
-        ])
+    fn to_doc(&self) -> String {
+        format!("if-let{}{}", &self.pattern.to_doc(), &self.expr.to_doc())
     }
 }
 
 impl Printable for Expr {
-    fn to_doc(&self) -> RcDoc<()> {
+    fn to_doc(&self) -> String {
         match self {
-            Expr::Term { sym, args, .. } => sexp(
-                // TODO(mbm): convenience for sexp with a fixed first element
-                Vec::from([sym.to_doc()])
-                    .into_iter()
-                    .chain(args.iter().map(|f| f.to_doc())),
-            ),
+            Expr::Term { sym, args, .. } => {
+                // sexp(
+                // Vec::from([sym.to_doc()])
+                //     .into_iter()
+                //     .chain(args.iter().map(|f| f.to_doc())),
+            }
             Expr::Var { name, .. } => name.to_doc(),
-            Expr::ConstInt { val, .. } => RcDoc::as_string(val),
-            Expr::ConstPrim { val, .. } => RcDoc::text("$").append(val.to_doc()),
+            Expr::ConstInt { val, .. } => format!("{}", val),
+            Expr::ConstPrim { val, .. } => format!("${}", val.to_doc()),
             Expr::Let { defs, body, .. } => {
-                let mut parts = Vec::new();
-                parts.push(RcDoc::text("let"));
-                parts.push(sexp(defs.iter().map(|d| d.to_doc())));
-                parts.push(body.to_doc());
-                sexp(parts)
+                let mut parts = String::from("let");
+                // Convert from a map to a string
+                parts.push_str(sexp(defs.iter().map(|d| d.to_doc())));
+                parts.push_str(&body.to_doc());
+                return parts
             }
         }
-    }
+    }   
 }
 
 impl Printable for LetDef {
-    fn to_doc(&self) -> RcDoc<()> {
-        sexp(vec![self.var.to_doc(), self.ty.to_doc(), self.val.to_doc()])
+    fn to_doc(&self) -> String {
+        format!("{}{}{}", &self.var.to_doc(), &self.ty.to_doc(), &self.val.to_doc())
```

Comment:
Conceptually, should be:
```
format!("({} {} {})", &self.var.to_doc(), &self.ty.to_doc(), &self.val.to_doc())
```

We'll likely want to represent sexprs separately later, but for now this works